deploy/api.yaml

```diff
@@ -82,8 +82,124 @@ paths:
         passthroughBehavior: "when_no_match"
         contentHandling: "CONVERT_TO_TEXT"
         type: "aws_proxy"
+  /token:
+    parameters:
+      - $ref: "#/components/parameters/GrantType"
+      - $ref: "#/components/parameters/Code"
+      - $ref: "#/components/parameters/RedirectUri"
+      - $ref: "#/components/parameters/ClientId"
+    post:
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              $ref: "#/components/schemas/TokenRequest"
+        required: true
+      responses:
+        "201":
+          description: "201 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TokenResponse"
+        "400":
+          description: "400 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: "500 response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+      x-amazon-apigateway-request-validator: " Validate token request"
+      x-amazon-apigateway-integration:
+        httpMethod: "POST"
+        uri:
+          Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${AccessTokenFunction.Arn}/invocations
+        responses:
+          default:
+            statusCode: "200"
+        passthroughBehavior: "when_no_match"
+        contentHandling: "CONVERT_TO_TEXT"
+        type: "aws_proxy"
 components:
+  parameters:
+    GrantType:
+      name: grant_type
+      in: query
+      required: true
+      description: Required value MUST be set to "authorization_code"
+      schema:
+        type: string
+    Code:
+      name: code
+      in: query
+      required: true
+      description: The authorization code received from the IPV Core
+      schema:
+        type: string
+    ClientId:
+      name: client_id
+      in: query
+      required: true
+      description: Required the value client_id has Maximum of 255 characters.
+      schema:
+        type: string
+    RedirectUri:
+      name: redirect_uri
+      in: query
+      required: true
+      description:  Required, and is the "redirect_uri" parameter was included in the authorization request
+      schema:
+        type: string
   schemas:
+    TokenRequest:
+      required:
+        - "grant_type"
+        - "code"
+        - "redirect_uri"
+        - "client_id"
+      type: "object"
+      properties:
+        grant_type:
+          type: "string"
+          pattern: "authorization_code"
+          example: "authorization_code"
+        code:
```

Comment:
If there's no minLength on the code does it mean API gateway will accept an empty value i.e.`...&code=&...`